submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestValidator.kt

```diff
@@ -0,0 +1,66 @@
+package ac.uk.ebi.biostd.submission.domain.request
+
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionRequest
+import ac.uk.ebi.biostd.persistence.common.service.RqtUpdate
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
+import ebi.ac.uk.base.orFalse
+import ebi.ac.uk.model.RequestStatus
+import ebi.ac.uk.model.RequestStatus.INVALID
+import ebi.ac.uk.model.RequestStatus.VALIDATED
+import ebi.ac.uk.security.integration.components.IUserPrivilegesService
+import mu.KotlinLogging
+import uk.ac.ebi.events.service.EventsPublisherService
+
+private val logger = KotlinLogging.logger {}
+
+class SubmissionRequestValidator(
+    private val userPrivilegesService: IUserPrivilegesService,
+    private val eventsPublisherService: EventsPublisherService,
+    private val queryService: SubmissionPersistenceQueryService,
+    private val requestService: SubmissionRequestPersistenceService,
+) {
+    suspend fun validateRequest(
+        accNo: String,
+        version: Int,
+        processId: String,
+    ) {
+        var requestStatus = INVALID
+
+        requestService.onRequest(accNo, version, RequestStatus.INDEXED_CLEANED, processId) {
+            requestStatus = validateRequest(it)
+            RqtUpdate(it.withNewStatus(requestStatus))
```

Comment:
I agree, this is better to handle in another ticket. Actually, this will be part of the async validation implementation. The aim of the current PR is just to avoid files modifications